pdns/zoneparser-tng.cc

```diff
@@ -272,6 +272,38 @@ pair<string,int> ZoneParserTNG::getLineNumAndFile()
     return {d_filestates.top().d_filename, d_filestates.top().d_lineno};
 }
 
+void fixTxtRecord(string &txt) {
+  bool in_quote = false;
+  bool needs_end_quote = false;
+
+  // Bind automatically puts quotes around text blocks that don't have them
+  for( string::size_type i = 0; i < txt.size(); ++i ) {
+    if( in_quote ) {
+      if( needs_end_quote && txt[i] == ' ' ) {
+        txt.insert(i, "\"");
+        ++i;
+        in_quote = false;
+      }
+      else if( !needs_end_quote && txt[i] == '"' ) {
+        in_quote = false;
+      }
+      else if( txt[i] == '\\' ) {     // escape - skip the next character
+        ++i;
+      }
+    } else if( txt[i] != ' ' ) {      // start of a new string
+      in_quote = true;
+      if( txt[i] != '"' ) {
+        txt.insert(i, "\"");
+        ++i;
```

Comment:
It looks like this will skip over the first character of an unquoted \<character-string>, which could be problematic if it is a backslash (e.g., `TXT \  2` should be recognized as identical to `TXT " " "2"`).